File: layrz_sdk/entities/ats_reception.py
"""Ats Reception entity"""

from datetime import datetime

from pydantic import BaseModel, Field

from layrz_sdk.constants import UTC


class AtsReception(BaseModel):
  """AtsReception entity"""

  pk: int = Field(description='Defines the primary key of the AtsReception')
  volume_bought: float = Field(
    description='Volume bought in liters',
    default=0.0,
  )
  real_volume: float | None = Field(
    description='Real volume in liters',
    default=None,
  )

  received_at: datetime = Field(
    description='Date and time when the reception was made',
    default_factory=lambda: datetime.now(UTC),
  )

File: go/entities/ats_reception.go
package entities

import "github.com/goldenm-software/layrz-sdk/go/v4/types"

// AtsReception represents a reception entity in the ATS module.
type AtsReception struct {
	// ID is the primary key of the reception.
	ID int64 `json:"id"`
	// VolumeBought is the volume bought in liters.
	VolumeBought float64 `json:"volume_bought"`
	// RealVolume is the real volume in liters.
	RealVolume *float64 `json:"real_volume,omitempty"`
	// ReceivedAt is the date and time when the reception was made.
	ReceivedAt types.UnixTime `json:"received_at"`
	// FuelType is the type of fuel used in the reception.
	FuelType string `json:"fuel_type"`
	// IsMerged indicates if the reception is merged with another.
	IsMerged bool `json:"is_merged"`
	// OrderID is the order ID associated with the reception.
	OrderID *int64 `json:"order_id,omitempty"`
}
